From c7720ca4c232d317cfc800a04eda4a1d5a44a944 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Wed, 1 Jul 2015 16:46:23 -0700
Subject: [PATCH] PARQUET-325: Always use row group size when padding is 0.

For block file systems, if the size left in the block is greater than
the max padding, a row group will be targeted at the remaining size.
However, when using 0 to turn padding off, the remaining bytes will
always be greater than padding and row groups can be targeted at very
tiny spaces. When padding is off, the next row group's size should
always be the default size.

Author: Ryan Blue <blue@apache.org>

Closes #234 from rdblue/PARQUET-325-padding-0-fix and squashes the following commits:

f4b3c2b [Ryan Blue] PARQUET-325: Always use row group size when padding is 0.
---
 .../java/org/apache/parquet/hadoop/ParquetFileWriter.java     | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java
index e2853763e0..5f265e0c1b 100644
--- a/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java
+++ b/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java
@@ -687,6 +687,10 @@ public void alignForRowGroup(FSDataOutputStream out) throws IOException {
 
     @Override
     public long nextRowGroupSize(FSDataOutputStream out) throws IOException {
+      if (maxPaddingSize <= 0) {
+        return rowGroupSize;
+      }
+
       long remaining = dfsBlockSize - (out.getPos() % dfsBlockSize);
 
       if (isPaddingNeeded(remaining)) {
